### PR TITLE
Fixing nodeCount to count all Basic Blocks including merged BBs

### PR DESCRIPTION
**Pull request by rasoolmaghareh**

### CODE REVIEW

**Comment by rasoolmaghareh** [comment]
@domainexpert Thank you for you r comment. I added TxTreeGraph::nodeCount. Also I have checked a few examples on Executor::transferToBasicBlock (with nodes ending with br and ret) and it seems to work fine. We may need to test other terminator instructions in future (can be seen in the table below):

terminators
.

File: lib/Core/TxTree.cpp [hunk 1]
```diff
@@ -1949,6 +1952,8 @@ std::string TxTree::getInterpolationStat() {
   std::stringstream stream;
   stream << "KLEE: done: Total reduced symbolic execution tree nodes = "
 		 << TxTreeGraph::nodeCount  << "\n";
+  stream << "KLEE: done: Total number of visited basic blocks = "
+		 << TxTree::blockCount  << "\n";
```
Comment (domainexpert) [inline lib/Core/TxTree.cpp]: @rasoolmaghareh TxTree:: is unnecessary here since this function belongs to TxTree.

File: lib/Core/Executor.cpp [hunk 2]
```diff
@@ -1483,6 +1483,7 @@ void Executor::transferToBasicBlock(BasicBlock *dst, BasicBlock *src,
     PHINode *first = static_cast<PHINode*>(state.pc->inst);
     state.incomingBBIndex = first->getBasicBlockIndex(src);
   }
+  TxTree::blockCount++;
```
Comment (domainexpert) [inline lib/Core/Executor.cpp]: @rasoolmaghareh I recommend enclosing this in `if (INTERPOLATION_ENABLED)` also as a visual indication that this is a part of Tracer-X.

File: lib/Core/TxTree.cpp [hunk 3]
```diff
@@ -1899,6 +1899,9 @@ bool TxTree::symbolicExecutionError = false;
 
 uint64_t TxTree::subsumptionCheckCount = 0;
 
+uint64_t TxTree::blockCount = 1;
+
+
```
Comment (domainexpert) [inline lib/Core/TxTree.cpp]: @rasoolmaghareh Extra blank line may not be necessary here.

**Comment by domainexpert** [comment]
@rasoolmaghareh Also thanks for the nice table! Where did you get it from?

**Comment by domainexpert** [comment]
@rasoolmaghareh If you did rebase, please push to this branch using:
```
git push -f
```

**Comment by domainexpert** [comment]
@rasoolmaghareh The reason of using `-f` option is because the commit history has changed, so we want to "force push" the changes to the repository.

**Comment by rasoolmaghareh** [comment]
Thanks for your comments @domainexpert. The table is from "Handbook of Open Source Tools"
By Sandeep Koranne page 276.

**Comment by rasoolmaghareh** [comment]
@domainexpert I have applied the changes.

File: lib/Core/TxTree.h [hunk 4]
```diff
@@ -708,6 +708,7 @@ class TxTree {
   static Statistic executeMemoryOperationTime;
   static double entryNumber;
   static double programPointNumber;
+  static uint64_t blockCount;
```
Comment (domainexpert) [inline lib/Core/TxTree.h]: @rasoolmaghareh Perhaps this should be declared after `subsumptionCheckCount`, following the initialization order in `TxTree.cpp`.

File: lib/Core/Executor.cpp [hunk 5]
```diff
@@ -1483,6 +1483,9 @@ void Executor::transferToBasicBlock(BasicBlock *dst, BasicBlock *src,
     PHINode *first = static_cast<PHINode*>(state.pc->inst);
     state.incomingBBIndex = first->getBasicBlockIndex(src);
   }
+  if (INTERPOLATION_ENABLED)
+	  //blockCount increased to count all visited Basic Blocks
```
Comment (domainexpert) [inline lib/Core/Executor.cpp]: @rasoolmaghareh The indentation here should be 4 spaces less. You may want to try to use `git clang-format .` to automatically standardize the format of the updates. To configure `git` to be able to use `clang-format` command, you can put the [git-clang-format script](https://llvm.org/svn/llvm-project/cfe/trunk/tools/clang-format/git-clang-format) somewhere in your PATH. To format your updates, first add all the updated files, then say `git clang-format .` You may need to execute this command several times until there is no more change.

**Pull request merged**